### PR TITLE
(feat) Add interstitial system admin page that links to non-core apps

### PR DESCRIPTION
**Pull request by hadijahkyampeire**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
- This PR adds a new admin config app which has a page where the user can navigate to different apps outside the core 3.x
- It also adds a new menu item to our `App switcher` named `System Administration` which links to the above-mentioned page.
- It also creates an `ExtensionSlot` for card links that all apps that want to add card links can use. I have currently added the `OCL` cardLink since they are in the same repo. 

## Usage e.g in the Form builder
- Create a component with such a link
```js
import React from "react";
import { useTranslation } from "react-i18next";
import { Layer, ClickableTile } from "@carbon/react";
import { ArrowRight } from "@carbon/react/icons";

const FormBuilderCardLink: React.FC = () => {
  const { t } = useTranslation();
  const header = t("manageForms", "Manage Forms");
  return (
    <Layer>
      <ClickableTile
        id={`clickable-tile-${header}`}
        href={`${window.spaBase}/form-builder`}
        target="_blank"
        rel="no-refferer"
      >
        <div>
          <div className="heading">{header}</div>
          <div className="content">{t("formBuilder", "Form Builder")}</div>
        </div>
        <div className="iconWrapper">
          <ArrowRight size={16} />
        </div>
      </ClickableTile>
    </Layer>
  );
};

export default FormBuilderCardLink;
```

And add an extension in `index.ts` like
```js
      {
        id: "system-administration-form-builder-card-link",
        slot: "system-admin-page-card-link-slot",
        load: getAsyncLifecycle(
          () => import("./form-builder-admin-card-link.component"),
          options
        ),
        online: true,
        offline: true,
      },
```

## Screenshots
Screenshot 2023-06-04 at 04 07 06

## Issue

*None.*
- https://issues.openmrs.org/browse/O3-2126

### CODE REVIEW

**Comment by hadijahkyampeire** [comment]
@ibacher please review my PR.

File: packages/esm-admin-systemconfig-app/src/dashboard/index.component.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export const SystemAdministrationDashbord = () => {
+  const { t } = useTranslation();
+
+  return (
+    <div className={styles.systemAdminPage}>
+      <div className={styles.breadcrumbsContainer}>
+        <ExtensionSlot extensionSlotName="breadcrumbs-slot" />
+      </div>
+      <div className={styles.cardsView}>
+        <LinkCard header={t('config', 'Configurations')} viewLink="/openmrs/admin">
```
Comment (ibacher) [inline packages/esm-admin-systemconfig-app/src/dashboard/index.component.tsx:18]: Ok, so the main comment here: This should be implemented as an extension slot with each app providing the link card as an extension.

Comment (hadijahkyampeire) [reply packages/esm-admin-systemconfig-app/src/dashboard/index.component.tsx:18]: @ibacher I have made it an ExtensionSlot and so far used it in the `OCL` app and its working. I will push other PRs for other repos.

**Comment by hadijahkyampeire** [comment]
@ibacher I am not sure why tests are failing, but they probably need a rerun but I am not able to trigger that.

**Comment by hadijahkyampeire** [comment]
Thanks @jayasanka-sack I have done that, hopefully the tests will pass.

File: packages/esm-system-admin-app/src/root.component.tsx [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
-import { SystemAdministrationDashbord } from './dashboard/index.component';
+import { SystemAdministrationDashboard } from './dashboard/index.component';
 
 const RootComponent: React.FC = () => {
   return (
     <BrowserRouter basename={`${window.spaBase}/system-administration`}>
       <Routes>
-        <Route path="/" element={<SystemAdministrationDashbord />} />
+        <Route path="/" element={<SystemAdministrationDashboard />} />
```
Comment (ibacher) [inline packages/esm-system-admin-app/src/root.component.tsx:10]: We don't actually need the path...

```suggestion
        <Route element={<SystemAdministrationDashboard />} />
```

File: packages/esm-system-admin-app/src/dashboard/index.component.tsx [hunk 3]
```diff
@@ -6,13 +6,13 @@ import { LinkCard } from './card.component';
...
   const { t } = useTranslation();
 
   return (
     <div className={styles.systemAdminPage}>
       <div className={styles.breadcrumbsContainer}>
-        <ExtensionSlot extensionSlotName="breadcrumbs-slot" />
+        <ExtensionSlot name="breadcrumbs-slot" />
       </div>
       <div className={styles.cardsView}>
         <LinkCard header={t('config', 'Configurations')} viewLink="/openmrs/admin">
```
[the remark this answers is not in the document]
Comment (ibacher) [reply packages/esm-system-admin-app/src/dashboard/index.component.tsx:18]: However, I'd actually just declare this as a normal extension with `order: 0`. That way, we allow implementations to move it around or disable it altogether if they so chose.

File: packages/esm-system-admin-app/src/dashboard/card.component.tsx [hunk 4]
```diff
@@ -16,10 +16,10 @@ export const LinkCard: React.FC<LinkCardProps> = ({ header, viewLink, children }
     <Layer>
       <ClickableTile
         className={styles.overviewCard}
-        id={`clickable-tile-${header}`}
         href={viewLink}
+        id={`clickable-tile-${header}`}
```
[the remark this answers is not in the document]
Comment (hadijahkyampeire) [reply packages/esm-system-admin-app/src/dashboard/card.component.tsx:20]: Sounds good, I will remove them for now.

**Review by ibacher — APPROVED** [review]

File: packages/esm-admin-openconceptlab-app/src/admin-ocl-card-link.component.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Layer, ClickableTile } from '@carbon/react';
+import { ArrowRight } from '@carbon/react/icons';
+
+const OpenConceptLabCardLink: React.FC = () => {
+  const { t } = useTranslation();
+  return (
+    <Layer>
```
Comment (ibacher) [inline packages/esm-admin-openconceptlab-app/src/admin-ocl-card-link.component.tsx:9]: At some point, we probably want to look into turning this into a component so we can reduce the per-app boiler-plate and ensure some kind of consistency.

**Comment by hadijahkyampeire** [comment]
Thanks @denniskigen and @ibacher

**Comment by ibacher** [comment]
Yes, otherwise it won't show up by default. Try to add it near the openconceptlab line (I'm trying to keep modules from monorepos near each other).

**Pull request merged**